### ISSUE TITLE
Dark and bright green dots change..


### ISSUE THREAD

**Issue by robertrub** [nightscout/AndroidAPS#3415]
I have changed transmitters and got the yellow triangle "recalculated data". Ok, I'm used to it.
The strange thing is that in the graphs, the dark and bright green dots change status/color often... How come ?
![Screenshot_20240825_140813_AAPS.jpg](https://github.com/user-attachments/assets/06ceb381-6f99-444c-ab4f-5bac3c68db96)

![Screenshot_20240825_142618_AAPS.jpg](https://github.com/user-attachments/assets/b5aba9c8-2752-47d8-a0f2-245459d2321c)

![Screenshot_20240825_165937_AAPS.jpg](https://github.com/user-attachments/assets/75474299-dfc5-4a2f-94f2-b227acad7690)

![Screenshot_20240825_170326_AAPS.jpg](https://github.com/user-attachments/assets/a52d701b-69f2-44aa-b7f1-59848f73620d)

**Comment by robertrub** [comment]
![Screenshot_20240825_172837_AAPS.jpg](https://github.com/user-attachments/assets/9fe850e8-37c7-4502-b608-b2ee5d2ddca2)

They are all bright now...

**Comment by robertrub** [comment]
And dark dots are back...
![Screenshot_20240825_173300_AAPS.jpg](https://github.com/user-attachments/assets/30278f71-3d93-4d0d-8d68-1020530d1a19)

**Comment by MilosKozak** [comment]
caused by delayed data

**Comment by robertrub** [comment]
@MilosKozak You closed this issue but the dots changing color was NOT due to missed readings. I had added the _missed readings arriving later_ info thinking it might help to give you a search direction. All my screenshots were done when the phone was near the G6 and xDrip was receiving the BGs regularly.

I think the problem comes from the smoothing in AAPS but I'm not sure.

I'm about to swap sensors. Maybe the desynchronisation of the 5 min interval between the 2 transmitters is creating a problem in the smoothing section... (very wild guess).

I'll stop smoothing before the swap and see if anything behaves differently...

Please reopen this issue as it is not solved.

**Comment by robertrub** [comment]
Stopping smoothing in AAPS does NOT stop the dot problem..The end of 36 hours after transmitter change stops the problem.

So, the "culprit" is with/around the recalculated data code (when there is a "slide" of the 5 min delays when the G6 transmitters are changes).

**Comment by vanelsberg** [comment]
> the dark and bright green dots change status/color often...
Hmmm. Using BYODA: do not see these different colors at all. You are using xDrip+?

**Comment by robertrub** [comment]
Yes, xDrip.

**Comment by vanelsberg** [comment]
Haven't used xDrip+ other then in follower mode. Could it be xDrip+ does some filling in (or Libre smoothing)?
Just asking....

**Comment by robertrub** [comment]
I saw once that, if missed 2 readings (far from phone), then on next connection, only reading 3 arrived and 5 min later, readings 1,2 and 4 arrived.

This said, the dark/light green dots (and they change color !!) happens even when I'm near the phone and all the BGs arrive every 5 min.

Look at the 2nd logs I posted on this issue. They are strange...

https://github.com/nightscout/AndroidAPS/issues/3413
